### ISSUE TITLE
'names from variables and functions must the same' error in summarizeResult

### ISSUE THREAD

**Issue by ablack3** [darwin-eu-dev/PatientProfiles#484]
In summarizeResult would it be possible to allow users to change the variables without changing the functions used to summarize. 

For example I do not need any dateVariables so I removed it from the variables list but then got an error becuase the functions list had a dateVariables element. 

I think if dateVariables does not exist in the variables argument then we can just ignore it and not throw an error.

**Comment by ablack3** [comment]
``` r
library(CDMConnector)
con <- DBI::dbConnect(duckdb::duckdb(), eunomia_dir())
cdm <- cdm_from_con(con, "main", "main")

cohort_set <- read_cohort_set(system.file("cohorts3", package = "CDMConnector"))

cdm <- generate_cohort_set(cdm, cohort_set)
#> ℹ Generating 5 cohorts
#> ℹ Generating cohort (1/5) - gibleed_all_end_1)
#> ✔ Generating cohort (1/5) - gibleed_all_end_1) [106ms]
#> 
#> ℹ Generating cohort (2/5) - gibleed_all)
#> ✔ Generating cohort (2/5) - gibleed_all) [57ms]
#> 
#> ℹ Generating cohort (3/5) - gibleed_default_with_descendants)
#> ✔ Generating cohort (3/5) - gibleed_default_with_descendants) [47ms]
#> 
#> ℹ Generating cohort (4/5) - gibleed_default)
#> ✔ Generating cohort (4/5) - gibleed_default) [44ms]
#> 
#> ℹ Generating cohort (5/5) - gibleed_end_1)
#> ✔ Generating cohort (5/5) - gibleed_end_1) [45ms]
#> 

library(PatientProfiles)

detectVariables(dplyr::collect(cdm$cohort), "date")
#> [1] "cohort_start_date" "cohort_end_date"


# this works with a warning
result <- PatientProfiles::summariseResult(
  cdm$cohort
)
#> Warning: ! `result_id` column is missing, please add it as it is a compulsory
#> column.

# this error seems unnecessary
result <- PatientProfiles::summariseResult(
  cdm$cohort,
  variables = list(dateVariables = c("cohort_start_date", "cohort_end_date"))
)
#> Error in `checkVariablesFunctions()`:
#> ! Names from variables and functions must be the same
#> Backtrace:
#>     ▆
#>  1. └─PatientProfiles::summariseResult(...)
#>  2.   └─PatientProfiles:::checkVariablesFunctions(variables, functions, table)
#>  3.     └─cli::cli_abort("Names from variables and functions must be the same")
#>  4.       └─rlang::abort(...)




sessionInfo()
#> R version 4.3.1 (2023-06-16)
#> Platform: aarch64-apple-darwin20 (64-bit)
#> Running under: macOS Sonoma 14.0
#> 
#> Matrix products: default
#> BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
#> LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
#> 
#> locale:
#> [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#> 
#> time zone: Europe/Amsterdam
#> tzcode source: internal
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] PatientProfiles_0.6.2 CDMConnector_1.3.0   
#> 
#> loaded via a namespace (and not attached):
#>  [1] jsonlite_1.8.8     dplyr_1.1.4        compiler_4.3.1     reprex_2.0.2      
#>  [5] tidyselect_1.2.0   stringr_1.5.1      blob_1.2.4         snakecase_0.11.1  
#>  [9] tidyr_1.3.1        yaml_2.3.8         fastmap_1.1.1      readr_2.1.5       
#> [13] R6_2.5.1           generics_0.1.3     knitr_1.45         backports_1.4.1   
#> [17] checkmate_2.3.1    tibble_3.2.1       R.cache_0.16.0     DBI_1.2.2         
#> [21] tzdb_0.4.0         pillar_1.9.0       R.utils_2.12.2     rlang_1.1.3       
#> [25] utf8_1.2.4         stringi_1.8.3      xfun_0.41          fs_1.6.3          
#> [29] omopgenerics_0.1.2 cli_3.6.2          withr_3.0.0        magrittr_2.0.3    
#> [33] digest_0.6.34      rJava_1.0-11       rstudioapi_0.15.0  dbplyr_2.4.0      
#> [37] hms_1.1.3          lifecycle_1.0.4    R.methodsS3_1.8.2  R.oo_1.25.0       
#> [41] vctrs_0.6.5        evaluate_0.23      glue_1.7.0         SqlRender_1.16.1  
#> [45] CirceR_1.3.2       duckdb_0.9.2-1     styler_1.10.2      fansi_1.0.6       
#> [49] rmarkdown_2.25     purrr_1.0.2        tools_4.3.1        pkgconfig_2.0.3   
#> [53] htmltools_0.5.7
cdmDisconnect(cdm)
```

<sup>Created on 2024-03-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by catalamarti** [comment]
this was addressed here #491
